sfge/src/test/java/com/salesforce/rules/unusedmethod/BaseUnusedMethodTest.java

```diff
@@ -255,25 +259,37 @@ protected void assertExpectations(
             String entryMethod,
             Collection<String> usedMethodKeys,
             Collection<String> unusedMethodKeys) {
-        TestUtil.buildGraph(g, sourceCodes);
 
+        //TODO: This method attempts to simulate what AbstractRuleRunner.runRules does minus all the thread specific code
+        //      We should refactor the AbstractRuleRunner so that the Thread Specific parts can be injected as a strategy
+        //      and then for testing we would inject a strategy that executes without threads so that we can simply
+        //      call through to the top level rule runner here in the tests. Doing this refactoring would help us
+        //      prevent accidentally forgetting to update this code when changes occur somewhere deep down
+        //      inside of the runner's logic (like calling postProcess for example).
+
+        TestUtil.buildGraph(g, sourceCodes);
         RemoveUnusedMethod rule = RemoveUnusedMethod.getInstance();
         MethodVertex entryMethodVertex =
                 TestUtil.getMethodVertex(g, entryDefiningType, entryMethod);
         PathBasedRuleRunner runner =
                 new PathBasedRuleRunner(g, Lists.newArrayList(rule), entryMethodVertex);
-        // Violations aren't actually generated during the `runRules()` call.
         List<Violation> violations = new ArrayList<>(runner.runRules());
+        violations.addAll(rule.postProcess(g)); // Important! This is where the violations are created for this rule
+
         UsageTracker usageTracker = new UsageTracker();
         for (String usedMethodKey : usedMethodKeys) {
             assertTrue(
                     usageTracker.isUsed(usedMethodKey),
                     "Expected usage of method " + usedMethodKey);
         }
+
+        Set<String> actualViolationKeys = violations.stream()
```

Comment:
Note that since a lot of tests use this helper... for now I just made it so that we can match the unusedMethodKeys to the violations by creating a new set of keys which I am calling actualViolationKeys. I am not sure why some unusedMethodKeys use @ vs # for the line number... so I simply normalized it with "::".

Reply:
Figured out why '#' was used instead of '@'... it was simply a wrongly written assertion in the ConstructorsTest class. Fixed (rebased and force pushed to update).